swarm/pss/notify/notify_test.go

```diff
@@ -0,0 +1,232 @@
+	adapter := adapters.NewSimAdapter(newServices(false))
+	net := simulations.NewNetwork(adapter, &simulations.NetworkConfig{
+		ID:             "0",
+		DefaultService: "bzz",
+	})
+	l_nodeconf := adapters.RandomNodeConfig()
+	l_nodeconf.Services = []string{"bzz", "pss"}
+	l_node, err := net.NewNodeWithConfig(l_nodeconf)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = net.Start(l_node.ID())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	r_nodeconf := adapters.RandomNodeConfig()
+	r_nodeconf.Services = []string{"bzz", "pss"}
+	r_node, err := net.NewNodeWithConfig(r_nodeconf)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = net.Start(r_node.ID())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = net.Connect(r_node.ID(), l_node.ID())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	l_rpc, err := l_node.Client()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	r_rpc, err := r_node.Client()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var l_addr string
+	err = l_rpc.Call(&l_addr, "pss_baseAddr")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var r_addr string
+	err = r_rpc.Call(&r_addr, "pss_baseAddr")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var l_pub string
+	err = l_rpc.Call(&l_pub, "pss_getPublicKey")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = r_rpc.Call(nil, "pss_setPeerPublicKey", l_pub, controlTopic, l_addr)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rsrcName := "foo.eth"
+	rsrcTopic := pss.BytesToTopic([]byte(rsrcName))
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*2)
+	defer cancel()
+	rmsgC := make(chan *pss.APIMsg)
+	r_sub, err := r_rpc.Subscribe(ctx, "pss", rmsgC, "receive", controlTopic)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer r_sub.Unsubscribe()
+	r_sub_update, err := r_rpc.Subscribe(ctx, "pss", rmsgC, "receive", rsrcTopic)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer r_sub_update.Unsubscribe()
+
+	updateMsg := []byte("xyzzy")
+	ctrl := NewController(psses[0])
+	ctrl.NewNotifier("foo.eth", 2, func(name string) ([]byte, error) {
+		msgSeq++
+		return updateMsg, nil
+	})
+
+	msg := NewMsg(MsgCodeStart, rsrcName, common.FromHex(r_addr))
+	smsg, err := rlp.EncodeToBytes(msg)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = r_rpc.Call(nil, "pss_sendAsym", l_pub, controlTopic, common.ToHex(smsg))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var inMsg *pss.APIMsg
+	select {
+	case inMsg = <-rmsgC:
+	case <-ctx.Done():
+		t.Fatal(ctx.Err())
+	}
+	dMsg := &Msg{}
+	err = rlp.DecodeBytes(inMsg.Msg, dMsg)
+	if err != nil {
+		t.Fatal(err)
+	} else if dMsg.GetName() != rsrcName {
+		t.Fatalf("expected name %s, got %s", rsrcName, dMsg.GetName())
+	} else if !bytes.Equal(dMsg.Payload[:len(updateMsg)], updateMsg) {
+		t.Fatalf("expected payload first %d bytes '%x', got '%x'", len(updateMsg), updateMsg, dMsg.Payload[:len(updateMsg)])
+	} else if len(updateMsg)+symKeyLength != len(dMsg.Payload) {
+		t.Fatalf("expected payload length %d, have %d", len(updateMsg)+symKeyLength, len(dMsg.Payload))
+	}
+
+	l_pssAddr := pss.PssAddress(common.FromHex(l_addr))
```

Comment:
Out of scope for this PR, but this could just be named `Address` as it is in the `pss` package.